plugins/Live/API.php

```diff
@@ -152,26 +161,34 @@ public function getLastVisitsForVisitor($visitorId, $idSite, $filter_limit = 10,
      */
     public function getLastVisitsDetails($idSite, $period = false, $date = false, $segment = false, $countVisitorsToFetch = false, $minTimestamp = false, $flat = false, $doNotFetchActions = false)
     {
-        if (false === $countVisitorsToFetch) {
-            $filter_limit  = Common::getRequestVar('filter_limit', 10, 'int');
-            $filter_offset = Common::getRequestVar('filter_offset', 0, 'int');
+        Piwik::checkUserHasViewAccess($idSite);
 
-            $countVisitorsToFetch = $filter_limit + $filter_offset;
+        if ($countVisitorsToFetch !== false) {
+            $filterLimit     = (int) $countVisitorsToFetch;
+            $filterOffset    = 0;
+        } else {
+            $filterLimit     = Common::getRequestVar('filter_limit', 10, 'int');
+            $filterOffset    = Common::getRequestVar('filter_offset', 0, 'int');
         }
 
         $filterSortOrder = Common::getRequestVar('filter_sort_order', false, 'string');
 
-        Piwik::checkUserHasViewAccess($idSite);
-        $dataTable = $this->loadLastVisitorDetailsFromDatabase($idSite, $period, $date, $segment, $countVisitorsToFetch, $visitorId = false, $minTimestamp, $filterSortOrder);
+        $dataTable = $this->loadLastVisitorDetailsFromDatabase($idSite, $period, $date, $segment, $filterOffset, $filterLimit, $minTimestamp, $filterSortOrder, $visitorId = false);
         $this->addFilterToCleanVisitors($dataTable, $idSite, $flat, $doNotFetchActions);
 
         $filterSortColumn = Common::getRequestVar('filter_sort_column', false, 'string');
-        $filterSortOrder  = Common::getRequestVar('filter_sort_order', 'desc', 'string');
 
         if ($filterSortColumn) {
-            $dataTable->queueFilter('Sort', array($filterSortColumn, $filterSortOrder));
+            $this->logger->warning('Sorting the API method "Live.getLastVisitDetails" by column is currently not supported. To avoid this warning remove the URL parameter "filter_sort_column" from your API request.');
         }
 
+        // Usually one would Sort a DataTable and then apply a Limit. In this case we apply a Limit first in SQL
+        // for fast offset usage see https://github.com/piwik/piwik/issues/7458. Sorting afterwards would lead to a
+        // wrong sorting result as it would only sort the limited results. Therefore we do not support a Sort for this
+        // API
+        $dataTable->disableFilter('Sort');
+        $dataTable->disableFilter('Limit'); // limit is already applied here
```

Comment:
We had to make sure not to apply the `Limit` filter again in GenericFilter. Otherwise, when requesting `limit 10, offset 25` we would return an empty result as we would fetch only 10 records here and it would apply the offset of 25 again in GenericFilter.

There were a few possibilities about how to make this work (preventing a filter from not to run as it was already applied here). I found this one as the most simple way and easiest to understand. Also other solutions would have rather ended in a hack. This way we make actually sure the Limit won't be applied.